src/app/core/cms/halremote.ts

```diff
@@ -12,7 +12,8 @@ export class HalRemote {
   constructor(
     private http: Http,
     private host: string,
-    private token: string
+    private token?: Observable<string>,
```

Comment:
This is the key here: the `token` is now an observable rather than static string.  As it might change down the road, if the old token expired.